garrysmod/lua/includes/extensions/vector.lua

```diff
@@ -3,10 +3,257 @@ local meta = FindMetaTable( "Vector" )
+	return v
+end
+
+--[[---------------------------------------------------------
+Converts the vector to bisector of two vectors
+Converts vector to result of a * |b| + b * |a|
+-----------------------------------------------------------]]
+function meta:Bisect( vec )
+	local na, nb = self:Length(), vec:Length()
+	self:Mul( nb )
+	self:Add( vec:GetMul( na ) )
+end
+
+--[[---------------------------------------------------------
+Returns a copy of bisector of two vectors
+-----------------------------------------------------------]]
+function meta:GetBisected( ... )
+	local v = Vector( self )
+	v:Bisect( ... )
+	return v
+end
+
+--[[---------------------------------------------------------
+Convets the vector to resized version of itself
+-----------------------------------------------------------]]
+function meta:Resize( num )
+	self:Normalize()
+	self:Mul( num )
+end
+
+--[[---------------------------------------------------------
+Returns a copy of the resized vector
+-----------------------------------------------------------]]
+function meta:GetResized( ... )
+	local v = Vector( self )
+	v:Resize( ... )
+	return v
+end
 
-	return Color( self.x * 255, self.y * 255, self.z * 255 )
+--[[---------------------------------------------------------
+Convets the vector to resized midpoint of itself
+-----------------------------------------------------------]]
+function meta:Midpoint( vec )
+	self:Add( vec )
+	self:Mul( 0.5 )
+end
 
+--[[---------------------------------------------------------
+Returns a copy of the midpoint vector
+-----------------------------------------------------------]]
+function meta:GetMidpoint( ... )
+	local v = Vector( self )
+	v:Midpoint( ... )
+	return v
+end
+
+--[[---------------------------------------------------------
+Returns a copy of the zero point vector
+-----------------------------------------------------------]]
+function meta:GetZero( ... )
+	local v = Vector( self )
+	v:Zero( ... )
+	return v
+end
+
+--[[---------------------------------------------------------
+Calculates 3x3 determinant by using other two vectors as rows
+-----------------------------------------------------------]]
+function meta:Determinant( vec2, vec3 )
+	local a, b, c = self:Unpack()
+	local d, e, f = vec2:Unpack()
+	local g, h, i = vec3:Unpack()
+	return ( ( a * e * i ) + ( b * f * g ) + ( d * h * c ) - ( g * e * c ) - ( h * f * a ) - ( d * b * i ) )
+end
+
+--[[---------------------------------------------------------
+Applies new basis to a vector by the three axises of an angle
+-----------------------------------------------------------]]
+function meta:Basis( ang )
+	local x = self:Dot( ang:Forward() )
+	local y = self:Dot( ang:Right() )
+	local z = self:Dot( ang:Up() )
+	self:SetUnpacked( x, y, z )
+end
+
+--[[---------------------------------------------------------
+Applies new basis to a vector by the three axises of an angle
+-----------------------------------------------------------]]
+function meta:GetBasised( ... )

```

Comment:
"Basised" isn't a word, should still be ``GetBasis``.

Reply:
Fixed. Thanks !